[DOCS] [Enterprise Search] Migrate ingest pipelines/ML docs

leemthompo opened the pull request:
URL previews

- https://elasticsearch_101156.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/ingest-pipeline-search.html
- https://elasticsearch_101156.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/nlp-example.html
- https://elasticsearch_101156.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/ingest-pipeline-search-inference.html


## `TODO` ✅ 

- [x]  Fix broken doclink
- [x]  Standardize id syntax 
- [x] Add back commented-out images
- [x]  Update `Enterprise Search` references
- [x]  Add content currently redirected to `[id="ingest-pipeline-search-inference-update-mapping"]`
- [x] Consider sub-dividing main page (`search-ingest-pipelines.asciidoc`)
  - Remove upgrading info? (Used collapsible)
  - Remove FAQ? (Use collapsible)

leemthompo committed:
WIP, port docs

- Update link syntax
- Update ids
- Fix n^n build failures :/
-

leemthompo committed:
Merge remote-tracking branch 'origin' into ent-search-ml-docs

leemthompo committed:
Fix id for doclink

leemthompo committed:
Let's try this on for size

leemthompo committed:
Idem

leemthompo committed:
Update attributes, Test image rendering

leemthompo committed:
Update image name

leemthompo committed:
Fix typo

leemthompo committed:
Update filename

leemthompo committed:
Add images, cleanup, standardize naming

leemthompo committed:
Tweak heading

elasticsearchmachine commented:
Pinging @elastic/es-docs (Team:Docs)

chriscressman commented on docs/reference/ingest/search-nlp-tutorial.asciidoc:
TOC indentation is off:

![Screenshot 2023-10-24 at 11-28-26 Tutorial Natural language processing (NLP) Elasticsearch Guide master Elastic](https://github.com/elastic/elasticsearch/assets/362928/5af93a76-ffa7-49db-ba1b-fed88f89a084)

chriscressman commented on docs/reference/ingest/search-inference-processing.asciidoc:
TOC indentation is off:

![Screenshot 2023-10-24 at 11-32-15 Inference processing Elasticsearch Guide master Elastic](https://github.com/elastic/elasticsearch/assets/362928/ec8fb406-ffc2-4a88-907e-651dff997089)

chriscressman commented on docs/reference/ingest/search-nlp-tutorial.asciidoc:
![Screenshot 2023-10-24 at 11-28-02 Tutorial Natural language processing (NLP) Elasticsearch Guide master Elastic](https://github.com/elastic/elasticsearch/assets/362928/c68fc800-6e1e-4713-97cd-155f26adfc3d)

I suggest placing the tutorial last in the navigation. Then there is a natural progression of information:

1. Manage ingest pipelines in the Content UI
2. Use the inference processing functions of those pipelines
3. See an end-to-end example

chriscressman commented on docs/reference/ingest/search-inference-processing.asciidoc:
```diff
@@ -0,0 +1,185 @@
+[[ingest-pipeline-search-inference]]
+=== Inference processing
+
+Starting in 8.5, *Search* creates ingestion pipelines by default to work with indices created for search use cases.
```
Try to re-frame the feature as an Elasticsearch feature by focusing on the specific UI rather than a concept of a "Search" area or solution. For example:

```suggestion
When you create an index through the **Content** UI, a set of default ingest pipelines are also created, including a ML inference pipeline.
```

chriscressman commented on docs/reference/ingest/search-ingest-pipelines.asciidoc:
```diff
@@ -0,0 +1,313 @@
+[[ingest-pipeline-search]]
+== Ingest pipelines in Search
+
+<<ingest, Ingest Pipelines>> are an Elasticsearch native tool which let you perform common transformations on data before storing it in Elasticsearch.
```
Because this used to live in the Enterprise Search docs, the introduction provides context for ingest pipelines. Now that this doc is within the ingest pipelines documentation, we may be able to assume that context and instead provide context on the Content UI.

For example, we could communicate:

- You can manage ingest pipelines through Elasticsearch APIs or Kibana UIs
- The Content UI is for creating and managing indices optimized for search use cases (non time series data)
- When managing indices through that UI, you may prefer to manage ingest pipelines there
- This doc tells you how to do that

chriscressman commented on docs/reference/ingest/search-ingest-pipelines.asciidoc:
```diff
@@ -0,0 +1,313 @@
+[discrete#ingest-pipeline-search-details-generic-reference]
+==== `ent-search-generic-ingestion` Reference
+
+As soon as *Search* starts, this pipeline is automatically available to use.
```
This sentence no longer relevant, now that Enterprise Search is no longer a dependency.

chriscressman commented on docs/reference/ingest/search-ingest-pipelines.asciidoc:
```diff
@@ -0,0 +1,313 @@
+[%collapsible%closed]
+=============
+
+*Q*: The Elasticsearch docs note that there are three types of ingest pipelines.
```
I suggest revising this, since we're in the Elasticsearch docs now.

leemthompo replied:
Yeah I collapsed the upgrade and faq sections but originally I was wondering if we could remove these entirely at this stage given we're in 8.11+

leemthompo replied:
@chriscressman I would vote that we remove the FAQ here I don't think it's very useful at this stage

chriscressman replied:
:+1: Sounds good to me. I agree the value is low.

chriscressman commented on docs/reference/ingest/search-ingest-pipelines.asciidoc:
```diff
@@ -0,0 +1,313 @@
+  In 8.5+, the <<ingest-pipeline-search-pipeline-settings, index setting to enable binary content>> is required *in addition* to the configurations mentioned in the {enterprise-search-ref}/crawler-managing.html#crawler-managing-binary-content[Elastic web crawler Guide].
+
+* `ent-search-generic-ingestion` - Since 8.5, Native Connectors, Connector Clients, and new (>8.4) Elastic web crawler indices will all make use of this pipeline by default.
+  A natural evolution from `app_search_crawler` and `ent_search_crawler`, this pipeline is designed to meet more generic needs across *Search*.
```
Rather than substituting _Search_ for _Enterprise Search_, we would ideally rewrite sentences to avoid the proper noun. For example, in this sentence I think we could just drop "across Search".

In other cases, we could refer to the Content UI, search use cases, or use other language to re-frame this away from the former Enterprise Search solution. Sorry, I should have caught that before we migrated this content, but I overlooked the use of the variable, which is used almost exclusively in this doc.

leemthompo replied:
No worries I needed a second pair of eyes here in any case because I was focused on getting a green build rather than the content per se.

chriscressman reviewed (commented):
I tried to identify issues related to the migration or the scope change. Overall, the migration looks good :+1:

leemthompo committed:
Cleanup, rewordings

- Modified introduction in `search-inference-processing.asciidoc`.
- Changed "Search connector" to "Elastic connector".
- Adjusted heading levels in `search-inference-processing.asciidoc`.
- Simplified ingest pipelines intro in `search-ingest-pipelines.asciidoc`.
- Edited ingest pipelines section for the *Content* UI.
- Reordered file inclusions in `search-ingest-pipelines.asciidoc`.
- Formatted inference pipeline creation into steps in `search-nlp-tutorial.asciidoc`.

leemthompo committed:
Lingering erroneousness

leemthompo committed:
Delete FAQ

chriscressman reviewed (approved):
:rocket:
